src/wsi/wsi_window.h

```diff
@@ -0,0 +1,134 @@
+#pragma once
+
+#include <windows.h>
+
+#include "wsi_monitor.h"
+
+#include "../vulkan/vulkan_loader.h"
+
+namespace dxvk::wsi {
+
+  /**
+    * \brief Impl-dependent state
+    */
+  struct DxvkWindowState {
+#ifdef DXVK_WSI_WIN32
+    LONG style   = 0;
+    LONG exstyle = 0;
+    RECT rect    = { 0, 0, 0, 0 };
+#endif
+  };
+
+  /**
+    * \brief The size of the window
+    * 
+    * \param [in] hWindow The window
+    * \param [out] pWidth The width (optional)
+    * \param [out] pHeight The height (optional)
+    */
+  void getWindowSize(
+          HWND      hWindow,
+          uint32_t* pWidth,
+          uint32_t* pWeight);
+
+  /**
+    * \brief Resize a window
+    * 
+    * \param [in] hWindow The window
+    * \param [in] pState The swapchain's window state
+    * \param [in] Width The new width
+    * \param [in] Height The new height
+    */
+  void resizeWindow(
+          HWND             hWindow,
+          DxvkWindowState* pState,
+          uint32_t         Width,
+          uint32_t         Height);
+
+  /**
+    * \brief Sets the display mode for a window/monitor
+    * 
+    * \param [in] hMonitor The monitor
+    * \param [in] hWindow The window (may be unused on some platforms)
+    * \param [in] pMode The mode
+    * \param [in] EnteringFullscreen Are we entering fullscreen?
+    * \returns \c true on success, \c false on failure
+    */
+  bool setWindowMode(
+          HMONITOR         hMonitor,
+          HWND             hWindow,
+    const WsiMode*         pMode,
+          bool             EnteringFullscreen);
```

Comment:
What is `EnteringFullscreen` needed for?

Reply:
(It was a poor abstraction for something, that I can do better)

Reply:
I don't like it either, which is why I was asking.